LLMonFHIR/FHIRInterpretation/UserStudy/UserStudyChatViewModel.swift

```diff
@@ -40,30 +39,24 @@ final class UserStudyChatViewModel: MultipleResourcesChatViewModel {  // swiftli
     }
 
     /// The current navigation state of the study
-    var navigationState: NavigationState { _navigationState }
+    private(set) var navigationState: NavigationState = .introduction
 
     /// Controls the visibility of the survey view
-    var isSurveyViewPresented: Bool { _isSurveyViewPresented }
+    private(set) var isSurveyViewPresented = false
 
     /// Controls the visibility of the dismiss confirmation dialog
-    var isDismissDialogPresented: Bool { _isDismissDialogPresented }
+    private(set) var isDismissDialogPresented = false
 
     /// Controls the visibility of the task instruction alert
-    var isTaskIntructionAlertPresented: Bool {
-        guard isTaskIntructionButtonDisabled else {
-            return _isTaskIntructionAlertPresented
-        }
-
-        return false
-    }
+    private(set) var isTaskIntructionAlertPresented = false
```

Comment:
If there is an error during the initial generation (e.g. put in a wrong API key in the settings, then start the chat) it doesn't show an alert and goes into a corrupted state where one can no longer exit the view (dismiss button doesn't work). 

I think it's due to the fact that the `TaskIntructionAlertPresented` is true at the time where the SpeziLLM module throws the error and would like to display an alert and switches the state into an error state. We should then first remove the task instructions alert, present the error alert, and ensure that the app is in a good state after the alert has been dismissed.